src/main/java/org/springframework/data/couchbase/repository/query/StringBasedN1qlQueryParser.java

```diff
@@ -103,7 +103,7 @@
 	private static final Logger LOGGER = LoggerFactory.getLogger(StringBasedN1qlQueryParser.class);
 	private final String statement;
 	private final QueryMethod queryMethod;
-	private final PlaceholderType placeHolderType;
+	private PlaceholderType placeHolderType;
```

Comment:
you sure this is non final now? if so it maybe shold be volatile

Reply:
It gets set when StringBasedN1qlQueryParser.checkPlaceHolders() is called by StringN1qlQueryCreator.getExpression(). To make it final, the code in getExpression() would need to be executed in the constructor. It can be done.  Not sure that it is better.  See next push.